drivers/serial/uart_nrfx_uarte.c

```diff
@@ -1189,6 +1189,13 @@ static void uarte_nrfx_set_power_state(struct device *dev, u32_t new_state)
 		assert(new_state == DEVICE_PM_LOW_POWER_STATE ||
 		       new_state == DEVICE_PM_SUSPEND_STATE ||
 		       new_state == DEVICE_PM_OFF_STATE);
+
+		/* Send stop events */
+		uarte->TASKS_STOPRX = 1;
+		uarte->TASKS_STOPTX = 1;
```

Comment:
that will trigger interrupt and in case of `CONFIG_UART_ASYNC_API` will result in event generation (TX_DONE or TX_ABOARTED).

Reply:
That sounds fine though, doesn't it? One might call these events spurious, but the driver should be equipped to handle those, no?

Reply:
TX is stopped at the end of every transmission, there is no need to stop it again. In asynchronous API, RX is also stopped so disabling UART is enough. Please check if following code fixes your issue:
```
#ifdef CONFIG_UART_ASYNC_API
		if (get_dev_data(dev)->async) {

			nrf_uarte_disable(uarte);
			return;
		}
#endif
		nrf_uarte_task_trigger(uarte, NRF_UARTE_TASK_STOPRX);
		while (!nrf_uarte_event_check(uarte, NRF_UARTE_EVENT_RXTO)) {}
		nrf_uarte_event_clear(uarte, NRF_UARTE_EVENT_RXTO);
```

Reply:
Nice, code is cleaner too. Force pushing.